spec/lib/api/v3/work_packages/work_package_representer_spec.rb

```diff
@@ -507,6 +505,39 @@
         end
       end
 
+      context 'when the user does not have the permission to remove watchers' do
+        before do
+          role.permissions.delete(:delete_work_package_watchers) and role.save
+        end
+
+        it 'should not have a link to add watcher' do
+          expect(subject).not_to have_json_path('_links/removeWatcher/href')
+        end
+      end
+
+      describe 'watchers link' do
+        context 'when the user is allowed to see watchers' do
+          it_behaves_like 'has an untitled link' do
```

Comment:
Specific tests are good.
I did not implement a title so I expect none.

If someone now decides to add a shitty title he will at least have to add a shitty spec too...